src/main/java/hudson/plugins/sshslaves/SSHConnector.java

```diff
@@ -259,8 +261,9 @@ public SSHConnector(int port, StandardUsernameCredentials credentials, String us
         this.prefixStartSlaveCmd = fixEmpty(prefixStartSlaveCmd);
         this.suffixStartSlaveCmd = fixEmpty(suffixStartSlaveCmd);
         this.launchTimeoutSeconds = launchTimeoutSeconds == null || launchTimeoutSeconds <= 0 ? null : launchTimeoutSeconds;
-        this.maxNumRetries = maxNumRetries == null || maxNumRetries <= 0 ? 0 : maxNumRetries;
-        this.retryWaitTime = retryWaitTime == null || retryWaitTime <= 0 ? 0 : retryWaitTime;
+        final Integer zero = 0;
+        this.maxNumRetries = maxNumRetries == null || maxNumRetries <= 0 ? zero : maxNumRetries;
+        this.retryWaitTime = retryWaitTime == null || retryWaitTime <= 0 ? zero : retryWaitTime;
```

Comment:
Actually, wouldn't the correct thing be to change the field type to `int` as that is the default value when `null` and that would remove the boxing/unboxing